### ISSUE TITLE
Call to resolveRouteBinding() does not match its new definition

### ISSUE THREAD

**Issue by mtvs** [laravel/framework#31946]
- Laravel Version: 7.0.#
- PHP Version: #.#.#
- Database Driver & Version:

After having trouble with the new custom field specification in route bindings, I found where the `resolveRouteBinding()` is called on a model and the caller only passes the `$value` parameter *without* passing anything for the `$field` parameter. 

https://github.com/laravel/framework/blob/bd67cc22af6fbc1f5745b0e3ce79e0a12b092048/src/Illuminate/Routing/RouteBinding.php#L70

**Comment by driesvints** [comment]
This isn't needed because `getRouteKeyName` will return the field.

**Comment by mtvs** [comment]
No, I'm talking about this new feature in L7 that allows the developer to specify a custom field in the route definition.

**Comment by driesvints** [comment]
Please provide full steps to reproduce.

**Comment by devcircus** [comment]
What isn't working for you. The new syntax works like it should.

**Comment by mtvs** [comment]
I doubt it works correctly. When I was updating [eloquent-hashids](https://github.com/mtvs/eloquent-hashids) for L7, I noticed that the `resolveRouteBinding()` that is overwriten there has changed and has a new parameter `$field` and I saw in the docs that there's a new feature for specifying a custom field in model bindings. So, the specified field should be passed to `$field`. I wrote the following test:

```php
	/**
	 * @test
	 */
	public function it_supports_specifying_a_field_in_the_route_binding()
	{
		$item = factory(Item::class)->create(['slug' => 'item-1']);

		Route::model('item', Item::class);

		Route::get('/item/{item:slug}', function ($resolved) use ($item) {
			$this->assertEquals($item->id, $resolved->id);
		})->middleware('bindings');

		$this->get("/item/item-1");
	}
```
but it fails with a model not found exception:

```
Illuminate\Database\Eloquent\ModelNotFoundException: No query results for model [Mtvs\EloquentHashids\Tests\Models\Item].

/home/mtvs/Code/eloquent-hashids/vendor/laravel/framework/src/Illuminate/Routing/RouteBinding.php:81
/home/mtvs/Code/eloquent-hashids/vendor/laravel/framework/src/Illuminate/Routing/Router.php:803
/home/mtvs/Code/eloquent-hashids/vendor/laravel/framework/src/Illuminate/Routing/Router.php:771
/home/mtvs/Code/eloquent-hashids/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:37
```
And I noticed that even though a custom field has been specified in the route but the `$field` still gets the `null` value.

**Comment by driesvints** [comment]
@mtvs please open an issue on their repo

**Comment by devcircus** [comment]
> I doubt it works correctly. 

The feature is widely used and works as expected.

**Comment by mtvs** [comment]
I found out why it generally works but won't work in that test. And it doesn't have anything to do with the package I mention before.

It works in implicit route bindings but not when a route parameter is explicitly bound to a model like in the previous test by calling `Route::model()`. That's because Laravel has 2 different calls to `resloveRouteBinding()`:

https://github.com/laravel/framework/blob/9e2522fb97feb06d55c1ffa65bb166fb5e24ee94/src/Illuminate/Routing/ImplicitRouteBinding.php#L45

https://github.com/laravel/framework/blob/9e2522fb97feb06d55c1ffa65bb166fb5e24ee94/src/Illuminate/Routing/RouteBinding.php#L70

So, it can still be an issue, though much less frequent.

I fixed the test in this way:

```php
		$given = factory(Item::class)->create(['slug' => 'item-1']);

		// Route::model('item', Item::class);

		Route::get('/item/{item:slug}', function (Item $item) use ($given) {
			$this->assertEquals($given->id, $item->id);
		})->middleware('bindings');

		$this->get("/item/item-1");
```
